### DISCUSSION TITLE
Reference List of Design Tokens

### DISCUSSION THREAD

**Discussion by jaeaton** [AvaloniaUI/Avalonia#16554] [Q&A]
I am trying to add some amount of consistency to my UI. Looking at Microsoft Fluent docs I found design tokens for [colors](https://react.fluentui.dev/?path=/docs/theme-colors--page). I am assuming that Avalonia has something similar (that the themes have default values for various features). Is there a reference list of tokens? It could also be that I am trying to use them wrong.

J

**Comment by stevemonaco** [comment]
After a quick glance, I don't think the same approach would be useful for Avalonia even if it were available. Besides being a lot of work to create+maintain, `ControlTheme`s are completely defined by the theme author so there are few guarantees across themes. This means you need to read the `ControlTheme` for the control you're trying to customize similar to reading an API you're trying to use.

To briefly explain, you have a hierarchy of colors and brushes so that specific colors can be replaced without affecting every other control.

* Define the Colors
    * https://github.com/AvaloniaUI/Avalonia/blob/2a9e7e9dcfc3e590bdd50b1e874ce70c0fafcc24/src/Avalonia.Themes.Fluent/Accents/BaseColorsPalette.xaml
* Define the general Brush resources
    * https://github.com/AvaloniaUI/Avalonia/blob/2a9e7e9dcfc3e590bdd50b1e874ce70c0fafcc24/src/Avalonia.Themes.Fluent/Accents/BaseResources.xaml
* Define the control-specific Brush resources
    * https://github.com/AvaloniaUI/Avalonia/blob/2a9e7e9dcfc3e590bdd50b1e874ce70c0fafcc24/src/Avalonia.Themes.Fluent/Accents/FluentControlResources.xaml
* Actually use the control-specific brush in a `ControlTheme`
    * https://github.com/AvaloniaUI/Avalonia/blob/2a9e7e9dcfc3e590bdd50b1e874ce70c0fafcc24/src/Avalonia.Themes.Fluent/Controls/Button.xaml#L75

As you can see from the specific brush name (`AccentButtonBackgroundPointerOver`) it would be difficult to text search for the desired brush. This is why I prefer to read the `ControlTheme` and actually understand how it works, especially because there are scenarios where the straightforward way to replace a resource will not work.

**Reply by jaeaton** [reply]
Thank you for the additional details. I am trying to wrap my head around this. I have a Grid Splitter that I currently have as crimson (as it is visible in both default and dark theme). I want it to match the rest of the UI and adjust with changes to dark or light theme.

Would I just us a setter for the grid splitter to SystemAccentColor?

**Reply by stevemonaco** [reply]
Only when you specifically want the OS's accent color (if available). Otherwise, replace the control-specific brush/color in the `ThemeDictionaries` for the specific theme variant. [See docs](https://docs.avaloniaui.net/docs/guides/styles-and-resources/how-to-use-theme-variants#defining-and-referencing-custom-variant-specific-resources)
